velox/functions/prestosql/fuzzer/CastFuzzerTest.cpp

```diff
@@ -0,0 +1,271 @@
+      return MatchType::EXACT_MATCH;
+    }
+
+    auto splitIntoMantissaExponent = [](std::string& val) {
+      auto pos = val.find('E');
+      return std::tuple(val.substr(0, pos - 1), val.substr(pos + 1));
+    };
+
+    auto [expectedMantissa, expectedExponent] =
+        splitIntoMantissaExponent(expectedString);
+    auto [actualMantissa, actualExponent] =
+        splitIntoMantissaExponent(actualString);
+
+    // Exponents should always be equal
+    VELOX_CHECK_EQ(actualExponent, expectedExponent);
+
+    if (expectedMantissa == actualMantissa) {
+      return MatchType::EXACT_MATCH;
+    }
+
+    // Handle case when the last significant digit is off or missing
+    auto smallestLength = expectedMantissa.length() < actualMantissa.length()
+        ? expectedMantissa.length()
+        : actualMantissa.length();
+
+    if (expectedMantissa.substr(0, smallestLength) ==
+        actualMantissa.substr(0, smallestLength)) {
+      return MatchType::LEAST_SIGNIFICANT_DIGIT_OFF;
+    }
+
+    // Note we dont handle cases like ..5999 and ..6000 where last n digits get
+    // rounded off
+    return MatchType::NOT_A_MATCH;
+  }
+
+  void evaluate(
+      exec::ExprSet& exprSet,
+      const RowVectorPtr& data,
+      const SelectivityVector& rows,
+      VectorPtr& result) {
+    exec::EvalCtx evalCtx(&execCtx_, &exprSet, data.get());
+    std::vector<VectorPtr> results{result};
+    exprSet.eval(rows, evalCtx, results);
+
+    if (!result) {
+      result = results[0];
+    }
+  }
+
+  RowVectorPtr makeRowVector(const TypePtr& type, vector_size_t size) {
+    VectorFuzzer::Options opts;
+    opts.vectorSize = size;
+    opts.nullRatio = 0;
+    VectorPtr input = vectorFuzzer_.fuzzFlat(type);
+    return vectorMaker_.rowVector(
+        std::vector<std::string>{std::string(kColName)},
+        std::vector<VectorPtr>{std::move(input)});
+  }
+
+  void seed(size_t seed) {
+    currentSeed_ = seed;
+    vectorFuzzer_.reSeed(seed);
+    rng_.seed(currentSeed_);
+  }
+
+  FuzzerGenerator rng_;
+  size_t currentSeed_{0};
+  std::unique_ptr<PrestoQueryRunner> referenceQueryRunner_;
+
+  std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::memoryManager()->addRootPool()};
+  std::shared_ptr<memory::MemoryPool> pool_{rootPool_->addLeafChild("leaf")};
+  std::shared_ptr<core::QueryCtx> queryCtx_{std::make_shared<core::QueryCtx>()};
+  core::ExecCtx execCtx_{pool_.get(), queryCtx_.get()};
+  facebook::velox::test::VectorMaker vectorMaker_{pool_.get()};
+  VectorFuzzer vectorFuzzer_{{}, pool_.get()};
+};
+
+} // namespace facebook::velox::exec::test
+
+int main(int argc, char** argv) {
+  using namespace facebook::velox;
+
+  ::testing::InitGoogleTest(&argc, argv);
+  folly::Init init(&argc, &argv);
+
+  memory::MemoryManager::initialize({});
+  dwrf::registerDwrfWriterFactory();
+
+  auto referenceQueryRunner = std::make_unique<exec::test::PrestoQueryRunner>(
+      FLAGS_presto_url,
+      "aggregation_fuzzer",
```

Comment:
No its not, If you change it to any other value, either the aggregation_fuzzer or this fails. I renamed it cast_fuzzer first, but what happens is that currently PrestoQueryRunner hard codes temporary tables to tmp and it will not allow other users apart from the one that created it to drop the table. This is a pain if you want to test both aggregation fuzzer and this fuzzer locally. 
The right place is to fix the PQR, but we can address that in another PR.